### PR TITLE
Provide special strategy for old Android webviews

### PR DESCRIPTION
**Pull request by crookedneighbor**
Some old android webviews had an issue where characters after a permacharacter could not be inserted. This was because in these environments, the input.value property had not been updated in the event loop when the listener gets called. By delaying the behavior, we get access to the value.

Delaying the formatting breaks it in regular Chrome, so we need a new strategy for these devices.

### CODE REVIEW

**Review by lilaconlee — APPROVED** [review]
Looks 👍  to me, but I don't know much about ancient android webview detection.

File: lib/device.js [hunk 1]
```diff
@@ -7,6 +7,8 @@ var isIos = require('browser-detection/is-ios');
 var isIE9 = require('browser-detection/is-ie9');
 
 var CHROME_REGEX = /Chrome/;
+// Old Android Webviews used specific versions of Chrome with 0.0.0 as their version suffix
```
Comment (quinnjn) [inline lib/device.js]: Might be worth documenting this version behavior is for KitKat Chromium based WebViews

https://developer.chrome.com/multidevice/user-agent#webview_user_agent

File: lib/strategies/old-android-chrome-based-webview.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+var AndroidChromeStrategy = require('./android-chrome');
+
+function OldAndroidWebviewStrategy(options) {
```
Comment (quinnjn) [inline lib/strategies/old-android-chrome-based-webview.js]: Not a huge fan of this name. Maybe:`KitKatChromiumBasedWebViewStrategy`

Comment (crookedneighbor) [inline lib/strategies/old-android-chrome-based-webview.js]: Yeah, I was struggling with the naming. I like this better.

File: lib/device.js (hunk 1 above)
Comment (intelliot) [inline lib/device.js]: Based on the link @quinnjn provided, I'm uneasy checking for the 0.0.0 suffix.

> If you’re attempting to differentiate between the WebView and Chrome for Android, you should look for the presence of the Version/_X.X_ string in the WebView user-agent string. Don’t rely on the specific Chrome version number (for example, 30.0.0.0) as the version numbers changes with each release.

Comment (crookedneighbor) [inline lib/device.js]: The major version changes with each release, but the `0.0.0` suffix does not.

Comment (crookedneighbor) [inline lib/device.js]: I'll include `Version\/\d\.\d* ` in the regex.

**Pull request merged**